Index: ruby-2.2.1/configure.in
===================================================================
--- ruby-2.2.1.orig/configure.in
+++ ruby-2.2.1/configure.in
@@ -4180,6 +4180,17 @@ else
 fi
 AC_SUBST(USE_RUBYGEMS)
 
+AC_ARG_ENABLE(versioned-paths,
+	AS_HELP_STRING([--disable-versioned-paths], [disable paths with version number]),
+	[enable_versioned_paths="$enableval"], [enable_versioned_paths=yes])
+if test x"$enable_versioned_paths" = xno; then
+    AC_DEFINE(DISABLE_VERSIONED_PATHS, 1)
+    USE_VERSIONED_PATHS=NO
+else
+    USE_VERSIONED_PATHS=YES
+fi
+AC_SUBST(USE_VERSIONED_PATHS)
+
 arch_hdrdir="${EXTOUT}/include/${arch}/ruby"
 AS_MKDIR_P("${arch_hdrdir}")
 config_h="${arch_hdrdir}/config.h"
Index: ruby-2.2.1/lib/rdoc/ri/paths.rb
===================================================================
--- ruby-2.2.1.orig/lib/rdoc/ri/paths.rb
+++ ruby-2.2.1/lib/rdoc/ri/paths.rb
@@ -12,9 +12,9 @@ module RDoc::RI::Paths
   version = RbConfig::CONFIG['ruby_version']
 
   BASE    = if RbConfig::CONFIG.key? 'ridir' then
-              File.join RbConfig::CONFIG['ridir'], version
+              File.join [RbConfig::CONFIG['ridir'], RbConfig::CONFIG['USE_VERSIONED_PATHS'] == 'YES' ? version : nil].compact
             else
-              File.join RbConfig::CONFIG['datadir'], 'ri', version
+              File.join [RbConfig::CONFIG['datadir'], 'ri', RbConfig::CONFIG['USE_VERSIONED_PATHS'] == 'YES' ? version : nil].compact
             end
 
   homedir = begin
Index: ruby-2.2.1/loadpath.c
===================================================================
--- ruby-2.2.1.orig/loadpath.c
+++ ruby-2.2.1/loadpath.c
@@ -40,11 +40,17 @@
 
 typedef char ruby_lib_version_string[(int)sizeof(RUBY_LIB_VERSION) - 2];
 
+#ifdef DISABLE_VERSIONED_PATHS
+#define RUBY_LIB                    RUBY_LIB_PREFIX
+#define RUBY_SITE_LIB2              RUBY_SITE_LIB
+#define RUBY_VENDOR_LIB2            RUBY_VENDOR_LIB
+#else
 #ifndef RUBY_LIB
 #define RUBY_LIB                    RUBY_LIB_PREFIX  "/"RUBY_LIB_VERSION
 #endif
 #define RUBY_SITE_LIB2              RUBY_SITE_LIB    "/"RUBY_LIB_VERSION
 #define RUBY_VENDOR_LIB2            RUBY_VENDOR_LIB  "/"RUBY_LIB_VERSION
+#endif
 #ifndef RUBY_ARCH_LIB_FOR
 #define RUBY_ARCH_LIB_FOR(arch)        RUBY_LIB         "/"arch
 #endif
@@ -70,8 +76,10 @@ const char ruby_initial_load_paths[] =
     RUBY_SITE_ARCH_LIB_FOR(RUBY_THINARCH) "\0"
 #endif
     RUBY_SITE_ARCH_LIB_FOR(RUBY_SITEARCH) "\0"
+#ifndef DISABLE_VERSIONED_PATHS
     RUBY_SITE_LIB "\0"
 #endif
+#endif
 
 #ifndef NO_RUBY_VENDOR_LIB
     RUBY_VENDOR_LIB2 "\0"
@@ -79,8 +87,10 @@ const char ruby_initial_load_paths[] =
     RUBY_VENDOR_ARCH_LIB_FOR(RUBY_THINARCH) "\0"
 #endif
     RUBY_VENDOR_ARCH_LIB_FOR(RUBY_SITEARCH) "\0"
+#ifndef DISABLE_VERSIONED_PATHS
     RUBY_VENDOR_LIB "\0"
 #endif
+#endif
 
     RUBY_LIB "\0"
 #ifdef RUBY_THINARCH
Index: ruby-2.2.1/tool/mkconfig.rb
===================================================================
--- ruby-2.2.1.orig/tool/mkconfig.rb
+++ ruby-2.2.1/tool/mkconfig.rb
@@ -44,6 +44,7 @@ continued_name = nil
 continued_line = nil
 install_name = nil
 so_name = nil
+path_version = "/$(ruby_version)"
 File.foreach "config.status" do |line|
   next if /^#/ =~ line
   name = nil
@@ -142,6 +143,8 @@ File.foreach "config.status" do |line|
     case name
     when "ruby_version"
       version = val[/\A"(.*)"\z/, 1]
+    when /^USE_VERSIONED_PATHS$/
+      path_version = nil if /NO/ =~ val
     end
   end
 #  break if /^CEOF/
Index: ruby-2.2.1/tool/rbinstall.rb
===================================================================
--- ruby-2.2.1.orig/tool/rbinstall.rb
+++ ruby-2.2.1/tool/rbinstall.rb
@@ -396,7 +396,7 @@ end
 
 install?(:doc, :rdoc) do
   if $rdocdir
-    ridatadir = File.join(CONFIG['ridir'], CONFIG['ruby_version'], "system")
+    ridatadir = File.join([CONFIG['ridir'], RbConfig::CONFIG['USE_VERSIONED_PATHS'] == 'YES' ? version : nil, "system"].compact)
     prepare "rdoc", ridatadir
     install_recursive($rdocdir, ridatadir, :mode => $data_mode)
   end
